src/app/js/formats/streamgraph/Streamgraph.js

```diff
@@ -0,0 +1,497 @@
+import React, { PureComponent } from "react";
+import compose from "recompose/compose";
+import { StyleSheet, css } from "aphrodite/no-important";
+import * as d3 from "d3";
+import {
+  zoomFunction,
+  distinctColors,
+  transformDataIntoMapArray,
+  getMinMaxValue,
+  cutStr
+} from "./utils";
+import injectData from "../injectData";
+import exportableToPng from "../exportableToPng";
+
+const styles = StyleSheet.create({
+  divContainer: {
+    overflow: "hidden",
+    position: "relative"
+  },
+  tooltip: {
+    position: "absolute"
+  },
+  vertical: {
+    marginTop: 60,
+    height: 190,
+    pointerEvents: "none"
+  },
+  legend: {
+    position: "relative",
+    columnCount: 3,
+    textAlign: "left",
+    marginLeft: 20,
+    paddingBottom: 30
+  },
+  legendItem: {
+    marginTop: 2
+  },
+  legendItemTooltip: {
+    visibility: "hidden",
+    backgroundColor: "#4e4e4e",
+    color: "#fff",
+    textAlign: "center",
+    borderRadius: "6px",
+    padding: "5px 0",
+
+    position: "absolute",
+    zIndex: 10,
+  },
+  legendItemText: {
+    marginLeft: 5,
+  },
+  legendButton: {
+    height: 15,
+    width: 15
+  }
+});
+
+class Streamgraph extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
I see no `this.setState` in this component. This, these values (width and height) are constants and don't needs to be store in state.